### ISSUE TITLE
Double normalization of already softmax-normalized features potentially distorting cluster assignments

### ISSUE THREAD

**Issue by pratheeshkumar99** [DanGuo1223/mzClustering#2]
Title: Inconsistency between paper and implementation in similarity calculation

Description:
In the paper, similarity between cluster assignments is defined as cosine similarity:
sij(yi, yj) = (yi · yj) / (||yi|| ||yj||)

However, current implementation:
1. Takes softmax outputs (already normalized)
2. Applies additional L2 normalization twice:
```python
features = F.normalize(features, p=2, dim=-1)
features = features / features.norm(dim=1)[:, None]


This creates two issues:
a) Unnecessary double normalization of already normalized vectors
b) Deviates from paper's cosine similarity calculation
Suggested Fix:
Since features are already probability distributions from softmax,

1. Use softmax outputs directly with cosine similarity


Would appreciate feedback on preferred approach.

2. Correct Implementation:
```python
# Get cluster probabilities
features = self.CLUST(x_p)  # Already softmax normalized

# Calculate cosine similarity directly
sim_mat = torch.matmul(features, torch.transpose(features, 0,1))
